packages/open-truss/src/configuration/engine-v1.tsx

```diff
@@ -27,21 +31,29 @@ export function engineV1(
   config: WorkflowV1,
   data: YamlType,
 ): RenderingEngine {
-  const renderFrames = (frames: FrameV1[]): ReactTree => {
-    return frames.map(({ view, data, frames: subFrame }, i) => {
-      const { component, props } = view
-      const Component = COMPONENTS[component]
-      return subFrame === undefined ? (
-        <Component key={i} data={data} config={config} {...props} />
-      ) : (
-        <Component key={i} data={data} config={config} {...props}>
-          {renderFrames(subFrame)}
-        </Component>
-      )
-    })
+  const renderFrames = async (frames: FrameV1[]): Promise<ReactTree> => {
+    return (() => {
```

Comment:
What does wrapping this in a function do?

Reply:
I had to do this to match the return type. `async` functions require the return type to be `Promise<>`. Before, the type was `Array<Promise<>>` which it did not like. Kind of awkward, but it seems to be working.